ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckerFileManager.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2023 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.replication;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.config.ReplicationConfig;
+import java.nio.file.StandardOpenOption;
+import java.util.EnumSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * File manager for BackupChecker, although it can be used for other purposes
```

Comment:
nit: if it can be used for other purposes, then let's rename this as something like `WritableFileManager` (writable because the only thing that is does is `writeToFile`. Also should it be an interface with a `default` implementation instead of being a class?